File: app/src/main/java/com/geniusforapp/movies/ui/base/BaseFragment.java
package com.geniusforapp.movies.ui.base;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;


import org.jetbrains.annotations.NotNull;

import dagger.android.support.DaggerFragment;

/**
 * Created by anajar on 5/15/17.
 */

public abstract class BaseFragment extends DaggerFragment {


    @Nullable
    @Override
    public View onCreateView(@NotNull LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(setContentView(), container, false);
    }

    @Override
    public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        bindView(view);
    }

    protected abstract int setContentView();

    protected abstract void bindView(View view);
}

File: app/src/main/java/com/geniusforapp/movies/ui/base/BaseFragment.kt
package com.geniusforapp.movies.ui.base


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import dagger.android.support.DaggerFragment

abstract class BaseFragment : DaggerFragment() {


    lateinit var activityContext: BaseActivity

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context != null) {
            activityContext = context as BaseActivity
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(setContentView(), container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        bindView(view)
    }

    protected abstract fun setContentView(): Int

    protected abstract fun bindView(view: View)

}
